time_unit/__init__.py

```
@@ -186,10 +238,31 @@ def __ge__(self, other):
+            return item.date_range
+        raise TypeError('Item {item!r} has no date range.')
+
+    def overlaps_with(self, item):
+        frm0, to0 = self._get_range(item)
+        frm, to = self.date_range
+        return to >= frm0 and to0 >= frm
+
+    def __contains__(self, item):
+        frm0, to0 = self._get_range(item)
+        frm, to = self.date_range
+        return frm <= frm and to0 <= to
```

Pylint (3.3.7)

[refactor] Redundant comparison - frm <= frm

(R0124)